core/types/calls.cc

```diff
@@ -307,6 +307,12 @@ TypePtr unwrapType(Context ctx, Loc loc, const TypePtr &tp) {
         return metaType->wrapped;
     }
     if (auto *classType = cast_type<ClassType>(tp.get())) {
+        if (classType->symbol.data(ctx)->derivesFrom(ctx, core::Symbols::OpusEnum())) {
+            // Opus::Enum instances are allowed to stand for themselves in type syntax positions.
+            // See the note in type_syntax.cc regarding Opus::Enum.
+            return tp;
+        }
+
```

Comment:
@DarkDimius I made `unwrapType` in calls.cc aware of Opus::Enum. I'm not sure if this is going too far.

Reply:
could you add the following test:
```
class Enum < Opus::Enum
  Entry = new
end

Enum::Entry.new # we should error here
T.cast(nil, T::Array[Enum::Entry]) # should be allowed
```